File: r/leap/leap.R
leap <- function(year) {
  divisible_by(year, 4) &&
    (
      divisible_by(year, 400) ||
      !divisible_by(year, 100)
    )
}

divisible_by <- function(year, number) {
  year %% number == 0
}

File: javascript/leap/leap.js
//
// This is only a SKELETON file for the "Leap" exercise. It's been provided as a
// convenience to get you started writing code faster.
//

var Year = function (input) {
  this.year = input
  this.divisible_by = function (number) {
    return this.year % number == 0
  };
};

Year.prototype.isLeap = function () {
  return (this.divisible_by(4) && !this.divisible_by(100))
    || this.divisible_by(400)
};

module.exports = Year;
